plugin/tasks.py

```diff
@@ -13,125 +13,264 @@
+        gcp_config['network'],
+        rules)
+    ctx.instance.runtime_properties[utils.TARGET_TAGS] = \
+        firewall[utils.TARGET_TAGS]
+    ctx.instance.runtime_properties[utils.SOURCE_TAGS] = \
+        firewall[utils.SOURCE_TAGS]
+    firewall = FirewallRule(gcp_config,
+                            ctx.logger,
+                            firewall,
+                            gcp_config['network'])
+    firewall.create()
+    ctx.instance.runtime_properties[utils.NAME] = firewall.name
+
+@operation
+@throw_cloudify_exceptions
+def delete_security_group(gcp_config, **kwargs):
+    if ctx.instance.runtime_properties.get(utils.TARGET_TAGS):
+        ctx.instance.runtime_properties.pop(utils.TARGET_TAGS)
+    if ctx.instance.runtime_properties.get(utils.SOURCE_TAGS):
+        ctx.instance.runtime_properties.pop(utils.SOURCE_TAGS)
+    delete_firewall_rule(gcp_config, **kwargs)
+
+
+@operation
+@throw_cloudify_exceptions
+def create_keypair(gcp_config,
+                   user,
+                   private_key_path,
+                   external,
+                   private_existing_key_path='',
+                   public_existing_key_path='',
+                   **kwargs):
+    keypair = KeyPair(gcp_config,
+                      ctx.logger,
+                      user,
+                      private_key_path)
+    if external:
+        keypair.private_key = ctx.get_resource(private_existing_key_path)
+        keypair.public_key = ctx.get_resource(public_existing_key_path)
+    else:
+        keypair.create()
+    keypair.add_project_ssh_key(user, keypair.public_key)
+    ctx.instance.runtime_properties['gcp_private_key'] = keypair.private_key
+    ctx.instance.runtime_properties['gcp_public_key'] = keypair.public_key
+    keypair.save_private_key()
+
+
+@operation
+@throw_cloudify_exceptions
+def save_private_key(gcp_config,
+                     private_key,
+                     user,
+                     private_key_path,
+                     **kwargs):
+    keypair = KeyPair(gcp_config,
+                      ctx.logger,
+                      user,
+                      private_key_path)
+    keypair.private_key = private_key
+
+
+@operation
+@throw_cloudify_exceptions
+def delete_private_key(gcp_config, user, private_key_path, **kwargs):
+    keypair = KeyPair(gcp_config,
+                      ctx.logger,
+                      user,
+                      private_key_path)
+    keypair.remove_private_key()
+
+
+@operation
+@throw_cloudify_exceptions
+def delete_keypair(gcp_config, user, private_key_path, **kwargs):
+    keypair = KeyPair(gcp_config,
+                      ctx.logger,
+                      user,
+                      private_key_path)
+    keypair.public_key = ctx.instance.runtime_properties['gcp_public_key']
+    keypair.remove_project_ssh_key()
+    ctx.instance.runtime_properties.pop('gcp_public_key')
+    ctx.instance.runtime_properties.pop('gcp_private_key')
+
```

Comment:
I think the private key should be deleted here, symmetric to how it is saved at the end of create_keypair